src/go/rdctl/cmd/shell.go

```diff
@@ -117,24 +127,59 @@ const restartDirective = "Either run 'rdctl start' or start the Rancher Desktop
 
 func checkLimaIsRunning(commandName string) bool {
 	// Ignore error messages; none are expected here
-	output, err := exec.Command(commandName, "ls", "0", "--format", "{{.Status}}").CombinedOutput()
+	cmd := exec.Command(commandName, "ls", "0", "--format", "{{.Status}}")
+	stderr := &cmdWriter{}
+	cmd.Stderr = stderr
+
+	stdout := &cmdWriter{}
+	cmd.Stdout = stdout
+
+	err := cmd.Start()
+	if err == nil {
+		err = cmd.Wait()
+	}
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Failed to run 'rdctl shell': %s\n", err)
 		return false
 	}
-	if strings.HasPrefix(string(output), "Running") {
+	if strings.HasPrefix(stdout.result, "Running") {
 		return true
 	}
-
-	if output == nil || strings.Contains(string(output), "No instance matching 0 found") {
+	stderrMsg := getMsgPart(stderr.result)
+	if strings.HasPrefix(stderrMsg, "No instance matching 0 found.") {
```

Comment:
I think the `getMsgPart` function depends too tightly on the implementation of the warning in `limactl` and can easily be broken by future Lima changes. I think the previous check using `strings.Contains` is more robust. Do you expect it to get false positives? How?

Reply:
I thought the problem was in the logger -- if the logger detects that we're emitting output to a non-tty, it wraps the message in a logger-typical payload. The `getMsgPart` looks to see if that's the case, and if it is, it pulls it out. I'll verify that it starts with a `time` field before pulling out a `msg` field.

Reply:
I also stopped emitting stderr if there's a state value. And I only call `getMsgPart` if I want to display the full contents written to `stderr`. Otherwise `strings.Contains` is fine.